app/views/admin/firms/show.html.erb

```diff
@@ -25,6 +25,16 @@
     <p>
       <strong>Added:</strong> <%= @firm.created_at.to_s(:long) %>
     </p>
+    <p class="t-approved">
+      <strong>Approved:</strong> <%= @firm.approved_at&.to_s(:long) || 'Not approved' %>
+    </p>
+    <p>
+      <%= button_to('Approve Firm',
+                    admin_firm_approve_path(@firm.id),
+                    class: 'btn btn-success t-approve-firm',
+                    method: :post,
+                    disabled: @firm.approved_at.present?) %>
+    </p>
```

Comment:
Would the ui work better if the button did not show at all if the firm has already been approved?

Reply:
Very good idea. Yes, going to change it.